src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectToolsPackageCommandResolver.cs

```diff
@@ -0,0 +1,205 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal class ProjectToolsPackageCommandResolver : ICommandResolver
+    {
+        private static readonly NuGetFramework s_toolPackageFramework = FrameworkConstants.CommonFrameworks.DnxCore50;
+
+        private List<string> _allowedCommandExtensions;
+        private IPackagedCommandSpecCreator _packagedCommandSpecCreator;
+
+        public ProjectToolsPackageCommandResolver(IPackagedCommandSpecCreator packagedCommandSpecCreator)
+        {
+            _packagedCommandSpecCreator = packagedCommandSpecCreator;
+
+            _allowedCommandExtensions = new List<string>() 
+            {
+                FileNameSuffixes.DotNet.DynamicLib
+            };
+        }
+
+        public CommandSpec Resolve(CommandResolverArguments commandResolverArguments)
+        {
+            return ResolveFromProjectTools(
+                commandResolverArguments.CommandName, 
+                commandResolverArguments.CommandArguments,
+                commandResolverArguments.ProjectDirectory);
+        }
+
+        public CommandSpec ResolveFromProjectTools(
+            string commandName, 
+            IEnumerable<string> args,
+            string projectDirectory)
+        {
+            var projectContext = GetProjectContextFromDirectory(projectDirectory, s_toolPackageFramework);
+
+            if (projectContext == null)
+            {
+                return null;
+            }
+
+            var toolsLibraries = projectContext.ProjectFile.Tools;
```

Comment:
We should probably create an Extension for lists called EmptyIfNull<T> that returns an empty Enumerable<T> when the value is null. With that, we could remove the if condition immediately below. This code would be much cleaner that way.